src/engine/strat_engine/engine.rs

```diff
@@ -85,23 +87,14 @@ impl StratEngine {
         let mut table = Table::default();
         let mut incomplete_pools = HashMap::new();
 
-        for (pool_uuid, devices) in &pools {
-            match StratPool::setup(&dm, *pool_uuid, devices) {
+        for (pool_uuid, devices) in pools {
+            match setup_pool(&dm, pool_uuid, &devices, &table) {
```

Comment:
`setup_pool` is doing the duplicate name check and returning an error if we already have the name.  With this code change the error will be consumed and a warning message emitted instead of bubbling up the error and causing the daemon to exit.  Is this the behavior we want?  You touch on this in the commit documentation. 

The source code documentation is incorrect for `StratEngine::initialize` with this change of behavior.

Reply:
One question I'm asking with this PR is "Is there a compelling reason to return an error in ```initialize()``` on this condition but not in ```block_evaluate()```?" I don't see that there is one and to make this distinction makes the code less maintainable.

Reply:
It's probably a good thing to keep the behavior the same and because `stratisd-init` shouldn't exit on errors like this in initramfs.  However, there was a comment with a todo about distinguishing between error types.  There is a difference between being unable to bring up a pool because `thin_check` or `thin_repair` tool is missing vs. one of the devices isn't present.  Do we want the same behavior for those errors too?

Reply:
Existing code does not distinguish among those causes and PR doesn't change that. I can not imagine a future where the code will not have to distinguish among those different causes and modify its actions accordingly. But, at present, no plan has been developed for that. When such a plan is developed the necessary information will be obtainable by making use of newly invented, non-generic errors returned from ```setup_pool```. This, though, is an extensive undertaking, which will involve modifications to ```ThinPool::Setup``` and other methods, as well as more work on the existing error hierarchy and possibly other internal error heirarchies. It's definitely beyond the scope of this PR.

Reply:
Created https://github.com/stratis-storage/stratisd/issues/854 for tracking this